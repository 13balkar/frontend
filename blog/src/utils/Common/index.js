import monthNames from '../../constants/monthNames';
const getDaySuffix = (day) => {
  if (day > 3 && day < 21) return 'th';
  switch (day % 10) {
    case 1: return 'st';
    case 2: return 'nd';
    case 3: return 'rd';
    default: return 'th';
  }
};

const getFormattedDate = (date) => {
  date = new Date(date);
  const day = date.getDate();
  const month = date.getMonth();
  const year = date.getFullYear();
  return `${day}${getDaySuffix(day)} ${monthNames[month]}/${year}`;
};

export default getFormattedDate;
