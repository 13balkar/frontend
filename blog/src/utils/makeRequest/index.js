import axios from 'axios';
import { BACKEND_URL } from '../../constants/apiEndPoints';

const makeRequest = async (apiEndPoint, dynamicConfig) => {
  const requestConfig = {
    url: `${BACKEND_URL}${apiEndPoint.url}`,
    method: apiEndPoint.method,
    ...dynamicConfig
  };
  const { data } = await axios(requestConfig);
  return data;
};

export default makeRequest;
