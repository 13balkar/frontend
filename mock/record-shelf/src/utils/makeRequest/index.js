import axios from 'axios';
import { BACKEND_URL } from '../../constants/apiEndPoints';

const makeRequest = async (apiEndPoint, dynamicConfig) => {
  const requestConfig = {
    url: `${BACKEND_URL}${apiEndPoint.url}`,
    headers: {
      authorization: 'Bearer QWlzaHdhcnlhIE4'
    },
    method: apiEndPoint.method,
    ...dynamicConfig
  };
  const { data } = await axios(requestConfig);
  return data.data;
};

export default makeRequest;
