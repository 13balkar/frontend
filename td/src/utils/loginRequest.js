import axios from 'axios';

const register = async (dynamicConfig, navigate) => {
  try {
    const requestConfig = {
      url: 'http://localhost:4000/user/create',
      method: 'post',
      ...dynamicConfig
    };
    const { data } = await axios(requestConfig);
    return data;
  } catch (error) {
    if (navigate) {
      const errorStatus = error.response?.status;
      if (errorStatus) {
        navigate(`/error/${errorStatus}`);
      } else {
        navigate('/error');
      }
    }
  }
};
const login = async (dynamicConfig, navigate) => {
  console.log('dynamicConfig', dynamicConfig);
  try {
    const requestConfig = {
      url: 'http://localhost:4000/user/login',
      method: 'post',
      ...dynamicConfig
    };
    const { data } = await axios(requestConfig);
    localStorage.setItem('token', data.token);
    localStorage.setItem('username', dynamicConfig.data.userName);
    navigate('/todo');
    // return data;
  } catch (error) {
    if (navigate) {
      const errorStatus = error.response?.status;
      if (errorStatus) {
        navigate(`/error/${errorStatus}`);
      } else {
        navigate('/error');
      }
    }
  }
};

export { register, login };
