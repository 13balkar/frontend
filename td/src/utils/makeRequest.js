import axios from 'axios';
import { BACKEND_URL } from '../constants/apiEndPoints';

const makeRequest = async (apiEndPoint, dynamicConfig, navigate) => {
  try {
    const requestConfig = {
      url: `${BACKEND_URL}${apiEndPoint.url}`,
      method: apiEndPoint.method,
      ...dynamicConfig
    };
    // console.log('requestConfig', requestConfig);
    const { data } = await axios(requestConfig);
    return data;
  } catch (error) {
    if (navigate) {
      const errorStatus = error.response?.status;
      if (errorStatus) {
        navigate(`/error/${errorStatus}`);
      } else {
        navigate('/error');
      }
    }
  }
};

export default makeRequest;
