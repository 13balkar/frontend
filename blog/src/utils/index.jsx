import getFormattedDate from './Common';
import makeRequest from './makeRequest';

export { getFormattedDate, makeRequest };
