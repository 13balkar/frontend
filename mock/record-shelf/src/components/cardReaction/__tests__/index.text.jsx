import React from 'react';
import { render, waitFor, screen, fireEvent } from '@testing-library/react';
import CardReaction from '..';
import { makeRequest } from '../../../utils';
jest.mock('../../../utils/makeRequest');
const mockData = {
  count: 14,
  like: false
};

describe('CardReaction', () => {
  describe('data fetching', () => {
    it('should display loading when data is being fetched', async () => {
      makeRequest.mockResolvedValue(mockData);
      expect(makeRequest).toHaveBeenCalledTimes(0);
      render(<CardReaction id='1' />);
      expect(makeRequest).toHaveBeenCalledTimes(1);
      await waitFor(() => {
        expect(screen.getByText('loading')).toBeInTheDocument();
      });
    });
    it('should display error when data fetching fails', async () => {
      makeRequest.mockRejectedValue(new Error('Err'));
      expect(makeRequest).toHaveBeenCalledTimes(0);
      render(<CardReaction id='1' />);
      expect(makeRequest).toHaveBeenCalledTimes(1);
      await waitFor(() => {
        expect(screen.getByText('Err')).toBeInTheDocument();
      });
    });
    it('should display data when data fetching succeeds', async () => {
      makeRequest.mockResolvedValue(mockData);
      expect(makeRequest).toHaveBeenCalledTimes(0);
      render(<CardReaction id='1' />);
      expect(makeRequest).toHaveBeenCalledTimes(1);
      await waitFor(() => {
        expect(screen.getByText('14')).toBeInTheDocument();
      });
    });
  });
  describe('logic', () => {
    it('should display heart-red when like is true', async () => {
      makeRequest.mockResolvedValue({ count: 12, like: true });
      expect(makeRequest).toHaveBeenCalledTimes(0);
      render(<CardReaction id='1' />);
      expect(makeRequest).toHaveBeenCalledTimes(1);
      await waitFor(() => {
        expect(screen.getByAltText('genre')).toHaveAttribute('src', '/assets/heart-red.svg');
      });
    });
    it('should display heart-gray when like is false', async () => {
      makeRequest.mockResolvedValue({ count: 12, like: false });
      expect(makeRequest).toHaveBeenCalledTimes(0);
      render(<CardReaction id='1' />);
      expect(makeRequest).toHaveBeenCalledTimes(1);
      await waitFor(() => {
        expect(screen.getByAltText('genre')).toHaveAttribute('src', '/assets/heart-gray.svg');
      });
    });
    it('should display the passed count when rendered for the first time', async () => {
      makeRequest.mockResolvedValue({ count: 12, like: false });
      expect(makeRequest).toHaveBeenCalledTimes(0);
      render(<CardReaction id='1' />);
      expect(makeRequest).toHaveBeenCalledTimes(1);
      await waitFor(() => {
        expect(screen.getByText('12')).toBeInTheDocument();
      });
    });
    it('should increment the count by 1 when clicked on heart grey', async () => {
      makeRequest.mockResolvedValue({ count: 12, like: false });
      expect(makeRequest).toHaveBeenCalledTimes(0);
      const { getByAltText } = render(<CardReaction id='1' />);
      expect(makeRequest).toHaveBeenCalledTimes(1);
      await waitFor(() => {
        expect(screen.getByText('12')).toBeInTheDocument();
      });
      makeRequest.mockResolvedValue({ count: 13, like: true });
      const heart = getByAltText('genre').parentElement;
      fireEvent.click(heart);
      await waitFor(() => {
        expect(screen.getByText('13')).toBeInTheDocument();
      });
    });
    it('should decrement the count by 1 when clicked on heart red', async () => {
      makeRequest.mockResolvedValue({ count: 12, like: true });
      expect(makeRequest).toHaveBeenCalledTimes(0);
      const { getByAltText } = render(<CardReaction id='1' />);
      expect(makeRequest).toHaveBeenCalledTimes(1);
      await waitFor(() => {
        expect(screen.getByText('12')).toBeInTheDocument();
      });
      makeRequest.mockResolvedValue({ count: 11, like: false });
      const heart = getByAltText('genre').parentElement;
      fireEvent.click(heart);
      await waitFor(() => {
        expect(screen.getByText('11')).toBeInTheDocument();
      });
    });
    it('should toggle the color of the heart when clicked', async () => {
      makeRequest.mockResolvedValue({ count: 12, like: true });
      expect(makeRequest).toHaveBeenCalledTimes(0);
      const { getByAltText } = render(<CardReaction id='1' />);
      expect(makeRequest).toHaveBeenCalledTimes(1);
      await waitFor(() => {
        expect(screen.getByAltText('genre')).toHaveAttribute('src', '/assets/heart-red.svg');
      });
      makeRequest.mockResolvedValue({ count: 11, like: false });
      const heart = getByAltText('genre').parentElement;
      fireEvent.click(heart);
      await waitFor(() => {
        expect(screen.getByAltText('genre')).toHaveAttribute('src', '/assets/heart-gray.svg');
      });
    });
    it('should call makeRequest with the correct arguments when clicked', async () => {
      makeRequest.mockResolvedValue({ count: 12, like: true });
      expect(makeRequest).toHaveBeenCalledTimes(0);
      const { getByAltText } = render(<CardReaction id='1' />);
      expect(makeRequest).toHaveBeenCalledTimes(1);
      await waitFor(() => {
        expect(screen.getByAltText('genre')).toHaveAttribute('src', '/assets/heart-red.svg');
      });
      makeRequest.mockResolvedValue({ count: 11, like: false });
      const heart = getByAltText('genre').parentElement;
      fireEvent.click(heart);
      await waitFor(() => {
        expect(screen.getByAltText('genre')).toHaveAttribute('src', '/assets/heart-gray.svg');
        expect(makeRequest).toHaveBeenCalledTimes(3);
      });
    });
  });
});
