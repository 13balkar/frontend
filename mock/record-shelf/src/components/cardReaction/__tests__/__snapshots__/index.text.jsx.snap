// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardReaction snapshot should render correctly 1`] = `
<DocumentFragment>
  <div>
    loading
  </div>
</DocumentFragment>
`;
